File: FormacaoPhp/PHPAvancOOP2/php/classes/abstratas/Funcionario.php
<?php

namespace classes\abstratas;

abstract class Funcionario
{
    public $nome;
    public $cpf;
    public $salario;

    const piso = 981.33;

    public function __construct($nome, $cpf, $salario = self::piso)
    {
        $this->nome = $nome;
        $this->cpf = $cpf;
        $this->salario = $salario;
    }

    public function __toString()
    {
        return json_encode($this);
    }

    abstract public function getBonificacao();

    final public function aumentarSalario()
    {
        $this->salario *= 1.5;
    }
}

File: FormacaoCSharp/FormacaoCSharp/Parte3/P3_ByteBank/Funcionarios/Funcionario.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace P3_ByteBank.Funcionarios
{

    class Funcionario
    {
        public static int TotalDeFuncionarios { get; private set; }
        public string Nome { get; set; }
        public string Cpf { get; private set; }
        public double Salario { get; protected set; }

        public Funcionario(string cpf, double salario)
        {
            Console.WriteLine("Funcionario Instanciado");
            Cpf = cpf;
            Salario = salario;
            TotalDeFuncionarios++;
        }

        public virtual void AumentarSalario()
        {
            Salario *= 1.1;
        }

        ~Funcionario() { Console.WriteLine("Destruido " + Nome); }

        public virtual double GetBonificacao() => Salario * 0.10;
    }
}